pkg/pipeline/datasource_pipeline_templates.go

```diff
@@ -0,0 +1,94 @@
+package pipeline
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type PipelineTemplate struct {
+	ID               int    `json:"id"`
+	Name             string `json:"name"`
+	Namespace        string `json:"namespace"`
+	SyntaxVersion    string `json:"syntaxVersion"`
+	TemplateSourceId int    `json:"templateSourceId"`
+	LatestSha        string `json:"latestSha"`
+}
+
+const pipelineTemplatesUrl = "pipelines/api/v1/templates"
+
+func pipelineTemplatesDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePipelineTemplatesRead,
+
+		Schema: map[string]*schema.Schema{
+			"templates": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"namespace": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"syntax_version": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"template_source_id": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"latest_sha": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePipelineTemplatesRead(d *schema.ResourceData, m interface{}) error {
+	templates := []PipelineTemplate{}
+	_, err := m.(*resty.Client).R().
+		SetResult(&templates).
+		Get(pipelineTemplatesUrl)
+	if err != nil {
+		return err
+	}
+
+	pipelineTemplates := make([]map[string]interface{}, 0)
+
+	for _, v := range templates {
+		template := make(map[string]interface{})
```

Comment:
You don't need make. And you also more specifically know your types here:

```go
pipelineTemplates = map[string]map[string]interface{}
```
this also seems entirely redundant to me

Reply:
Can you clarify what you'd like to see here?  I'm not sure on this one.

Reply:
remove `make()` and just declare the map both inside and outside the loop. Make creates a pointer. I don't believe this is needed

Also, you declare `map[string]interface{}` but you actually know what you specifically have (it isn't `interface{}` at the top level)

```go
pipelineTemplates = []map[string]map[string]interface{}
template := map[string]interface{}
```
But the more I look at this the more I feel like you should be using a specific type and not just maps of maps